src/common/src/types/jsonb.rs

```diff
@@ -210,6 +210,12 @@ impl From<Value> for JsonbVal {
     }
 }
 
+impl<'a> From<&'a Value> for JsonbRef<'a> {
```

Comment:
It is an anti-pattern to
* take `serde_json::Value` out of `JsonbRef`
* process `serde_json::Value`
* put processed `serde_json::Value` back into a `JsonbVal`

because we may switch away from `serde_json` #11805. This pattern is a leak of implementation detail.

However, this means moving the concat implementation to `common` crate, which could hurt build time #12485

cc @wangrunji0408 @xxchan It's up to you how we proceed with the code organization.

Reply:
Since the complexity is not very high, I feel comfortable to proceed in whatever way. 🤣